server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/NodeReplacementAllocationDecider.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.cluster.routing.allocation.decider;
+
+import org.elasticsearch.cluster.metadata.IndexMetadata;
+import org.elasticsearch.cluster.metadata.Metadata;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.cluster.routing.RoutingNode;
+import org.elasticsearch.cluster.routing.ShardRouting;
+import org.elasticsearch.cluster.routing.allocation.RoutingAllocation;
+
+public class NodeReplacementAllocationDecider extends AllocationDecider {
+
+    public static final String NAME = "node_replacement";
+
+    private static final Decision NO_REPLACEMENTS = Decision.single(Decision.Type.YES, NAME,
+        "no node replacements are currently ongoing, allocation is allowed");
+
+    @Override
+    public Decision canAllocate(ShardRouting shardRouting, RoutingNode node, RoutingAllocation allocation) {
+        final Metadata metadata = allocation.metadata();
+        if (replacementOngoing(metadata) == false) {
+            return NO_REPLACEMENTS;
```

Comment:
Detail: I wonder if we need this distinction in the messages? A replace in a distant data tier seems largely irrelevant to highlight in the allocation explain API?

Reply:
I always favor as much distinction as possible, to aid in debugging. That's why I decided to make this one a separate message. If you think it's distracting I can remove it.

Reply:
I can live with it. I do worry about some minor confusion though. I wonder if we could make the messages very similar something like:

```
neither the source nor target node are part of an ongoing node replacement
neither the source nor target node are part of an ongoing node replacement (no replacements)
```

such that it is sort of the same message to users but the situation of any ongoing replacement or not is revealed.

This is really nit-picking though and I can be persuaded to leave this as is.